### ISSUE TITLE
Return proper error message when there is a TLS issue

### ISSUE THREAD

**Issue by vonHaupt** [sendgrid/sendgrid-csharp#963]
Trying to use the C# API (current version) and when ever we promote to the hosting service, no mail is sent and we get this error;

An error occurred while sending the request

Works just fine on my development client locally. Any idea of what could be blocked on the server that stops mail from going through? Here's the complete error message with a stack trace;

SENDGRID MAIL: CreateSingleEmailToMultipleRecipient, MESSAGE: An error occurred while sending the request. STACK TRACE: at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at SendGrid.Helpers.Reliability.RetryDelegatingHandler.d__4.MoveNext() in C:\Users\elmer\Documents\GitHub\sendgrid-csharp\src\SendGrid\Reliability\RetryDelegatingHandler.cs:line 56 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Net.Http.HttpClient.d__58.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at SendGrid.SendGridClient.d__24.MoveNext() in C:\Users\elmer\Documents\GitHub\sendgrid-csharp\src\SendGrid\SendGridClient.cs:line 194 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at SendGrid.SendGridClient.d__25.MoveNext() in C:\Users\elmer\Documents\GitHub\sendgrid-csharp\src\SendGrid\SendGridClient.cs:line 245 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at SendGrid.SendGridClient.d__26.MoveNext() in C:\Users\elmer\Documents\GitHub\sendgrid-csharp\src\SendGrid\SendGridClient.cs:line 256 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at CVPW.Web.Models.SMTPSendGridMgr.d__8.MoveNext() in P:\SourceCode-2018\CVPW-MVC\CVPW-Mvc\CVPW.Web\Models\SMTPSendGridMgr.cs:line 115

**Comment by vonHaupt** [comment]
Fixed this (with no help from the developer). It was a TLS issue. Would have been a lot easier if we could get proper error messages.

**Comment by thinkingserious** [comment]
Hello @vonHaupt,

Thank you for reporting this and the fix! I've updated the title to reflect the next action on this issue.

With best regards,

Elmer

**Comment by thinkingserious** [comment]
Since there has been no activity on this issue since March 1, 2020, we are closing this issue. Please feel free to reopen or create a new issue if you still require assistance. Thank you!
